support/util/imagemetadata.go

```diff
@@ -27,12 +28,62 @@ import (
 
 const cacheTTL = time.Hour * 12
 
+// MirrorAvailabilityCache caches mirror availability results to prevent flapping
+type MirrorAvailabilityCache struct {
+	cache map[string]mirrorCacheEntry
+	mutex sync.RWMutex
+}
+
+type mirrorCacheEntry struct {
+	available bool
+	timestamp time.Time
+	ttl       time.Duration
+}
+
 var (
+	// Global cache instance for mirror availability
+	mirrorCache = &MirrorAvailabilityCache{
+		cache: make(map[string]mirrorCacheEntry),
+	}
 	imageMetadataCache = cache.NewLRUExpireCache(1000)
 	manifestsCache     = cache.NewLRUExpireCache(1000)
 	digestCache        = cache.NewLRUExpireCache(1000)
 )
 
+// get checks if a mirror URL is cached and not expired
+func (c *MirrorAvailabilityCache) get(mirrorURL string) (bool, bool) {
+	c.mutex.RLock()
+	defer c.mutex.RUnlock()
+
+	if entry, exists := c.cache[mirrorURL]; exists {
+		if time.Since(entry.timestamp) < entry.ttl {
+			return entry.available, true // cache hit
+		}
+		// Entry expired, remove it
+		delete(c.cache, mirrorURL)
```

Comment:
Did you consider https://github.com/openshift/hypershift/pull/7184/files#r2498357437?